packages/sls-env/src/dependencies.ts

```diff
@@ -70,22 +95,33 @@ const resolveEventDependencies = <D, P, E, C>(
   context: C
 ): AppDependencyConverter<D> => {
   const updatedDependencies = Object.entries(dependencies).reduce((acc, [dependencyName, dependency]) => {
-    if (dependency?.type === 'EventDependency') {
-      const key = dependency.getKey(payload, event, context)
-      const eventDependency = dependency.dependencies[key]
-      if (!eventDependency) {
-        logger.warn(`could not extract event based dependency for ${dependencyName}`)
-        throw new EventDependencyError(`No event based dependency found for '${dependencyName}'`)
-      } else {
-        logger.debug(`replacing '${dependencyName}' with event based dependency`)
-        return { ...acc, [dependencyName]: eventDependency }
-      }
+    switch (dependency.type) {
+      case DependencyType.EventDependency:
+        const key = dependency.getKey(payload, event, context)
+        const eventDependency = dependency.dependencies[key]
+        if (!eventDependency) {
+          logger.warn(`could not extract event based dependency for ${dependencyName}`)
+          throw new EventDependencyError(`No event based dependency found for '${dependencyName}'`)
+        } else {
+          logger.debug(`replacing '${dependencyName}' with event based dependency`)
+          return { ...acc, [dependencyName]: eventDependency }
+        }
+      case DependencyType.DependencyFactory:
+        logger.debug(`replacing '${dependencyName}' with factory based dependency`)
+        return { ...acc, [dependencyName]: dependency.dependencyFactory({ payload, event, context }) }
```

Comment:
ideally we would move this out to separate function

Reply:
There was some overlap already with the types being used and the dependencies being resolved within `resolveEventDependencies`

I personally feel it is more readable within one file as a person reading the code could reference the types and functionality without having to switch between files.

I can move to another file if you think that's needed though